packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -124,14 +130,38 @@ export const TranslationSidebar = ({
   } ${chapterNumber}:${verseNumber}`;
 
   const lexiconEntryRef = useRef<HTMLDivElement>(null);
+  const [previewElement, setPreviewElement] = useState<HTMLDivElement | null>(
+    null
+  );
+  const [previewVerseIds, setPreviewVerseIds] = useState<string[]>([]);
 
   useEffect(() => {
     const { current } = lexiconEntryRef;
     // Highlight references to the currently selected verse
     current
       ?.querySelectorAll(`a[data-ref="${bdbCurrentVerseRef}"]`)
       .forEach((element) => element.classList.add('bg-yellow-300'));
-  }, [bdbCurrentVerseRef, lexiconEntry]);
+
+    current?.querySelectorAll<HTMLAnchorElement>(`a.ref`).forEach((element) => {
+      element.onclick = () => {
```

Comment:
We should be able to add a single click event on the wrapper element, and then check the target to see if it an anchor element before executing the click handler logic. Then we don't have to constantly add and remove these events on each individual anchor

Reply:
done